tools/rosconsole/src/rosconsole/rosconsole.cpp

```diff
@@ -425,6 +435,25 @@ void initialize()
     backend::function_notifyLoggerLevelsChanged = notifyLoggerLevelsChanged;
     backend::function_print = _print;
 
+    char* line_buffered = NULL;
+#ifdef _MSC_VER
+    _dupenv_s(&line_buffered, NULL, "ROSCONSOLE_STDOUT_LINE_BUFFERED");
+#else
+    line_buffered =  getenv("ROSCONSOLE_STDOUT_LINE_BUFFERED");
+#endif
+    if (line_buffered)
+    {
+      try
+      {
+        g_force_stdout_line_buffered = boost::lexical_cast<bool>(std::string(line_buffered));
```

Comment:
Regarding the lexical cast. Similar existing checks don't bother with it and just check for a fixed value, e.g.:

* https://github.com/ros/ros_comm/blob/148178b3e347d75719f1eab0f7fc38dacfa2c647/clients/roscpp/src/libros/init.cpp#L319-L326
* https://github.com/ros/ros_comm/blob/148178b3e347d75719f1eab0f7fc38dacfa2c647/tools/rosgraph/src/rosgraph/network.py#L244
* https://github.com/ros/ros_comm/blob/148178b3e347d75719f1eab0f7fc38dacfa2c647/tools/roslaunch/src/roslaunch/remoteprocess.py#L106-L107

And in case the value doesn't match the expectation they silently apply the default.